lms/djangoapps/courseware/tests/test_courses.py

```diff
@@ -132,6 +132,30 @@ def _fake_get_value(value, default=None):
                 all(course.org == alternate_course.org for course in microsite_courses)
             )
 
+    def test_get_courses_with_filter(self):
+        """
+        Verify that filtering performs as expected.
+        """
+        user = UserFactory.create()
+        non_mobile_course = CourseFactory.create(emit_signals=True)
+        mobile_course = CourseFactory.create(mobile_available=True, emit_signals=True)
+
+        test_cases = (
+            (None, {non_mobile_course.id, mobile_course.id}),
+            (dict(mobile_available=True), {mobile_course.id}),
+            (dict(mobile_available=False), {non_mobile_course.id}),
+        )
+        for filter_, expected_courses in test_cases:
```

Comment:
I could have possibly used `ddt` here.  The reasons I didn't were:

1. readability - if I used ddt, I would have somehow needed to pass in the set of courses that were expected, without having access to the courses within the ddt.
2. unit test performance time - since each run would require re-creating the courses (although this could have been addressed by updating this test class to use `SharedModulestoreTestCase` instead.

By the way, I did try to add an informative error message to the `self.assertEqual` call below so it's clear which test case failed, in case of an error.